### PR TITLE
external status code of response template takes precedence

### PR DESCRIPTION
**Pull request by struan**
Order the response templates query by external status code so if you
have an update with an external status code the response template
matching that is always used. Previously with the database left to
determine the order it could pick a response template with a matching
state rather than a matching external status.

### CODE REVIEW

File: t/open311/getservicerequestupdates.t [hunk 1]
```diff
@@ -1009,6 +1007,83 @@ foreach my $test ( {
...
+    title => "Acknowledgement 1",
+    text => "Thank you for your report. We will provide an update within 48 hours.",
+    auto_response => 1,
+    state => "in progress"
+});
+
+for my $test (
+    {
+        external_code => '090',
+        description => 'check numeric external status code in responset template override state',
```
Comment (dracos) [inline t/open311/getservicerequestupdates.t]: typo in description, and one underneath too

File: perllib/Open311/GetServiceRequestUpdates.pm [hunk 2]
```diff
@@ -262,17 +262,22 @@ sub comment_text_for_request {
...
     my $template;
     if ($state_changed || $ext_code_changed) {
+        my $order;
         my $state_params = {
             'me.state' => $state
         };
         if ($ext_code) {
             $state_params->{'me.external_status_code'} = $ext_code;
+            # make sure that empty string comes last. Can't use DESC as that orders
+            # empty string before codes starting with 0 and explicit ASC will order
```
Comment (dracos) [inline perllib/Open311/GetServiceRequestUpdates.pm]: This sort works fine, but I don't understand the comment - DESC sorts "0" before "" as far as I can see?

```
bci=> select state,external_status_code,external_status_code is null from response_templates
where auto_response order by external_status_code desc nulls last;
...
                  | 0015                 | f
                  | 0010                 | f
                  | 0010                 | f
investigating     |                      | f
duplicate         |                      | f
action scheduled  |                      | f
```

Comment (struan) [reply perllib/Open311/GetServiceRequestUpdates.pm]: weird, I definitely had issues with this when I was testing. I will see if I can work out what's going on.

Comment (struan) [reply perllib/Open311/GetServiceRequestUpdates.pm]: hmmm, I have no idea why it was failing like that for me :|

File: perllib/Open311/GetServiceRequestUpdates.pm [hunk 3]
```diff
@@ -262,17 +262,22 @@ sub comment_text_for_request {
...
+            # empty string before codes starting with characters
+            $order = { order_by => \"me.external_status_code NULLS LAST" };
```
Comment (dracos) [inline perllib/Open311/GetServiceRequestUpdates.pm]: You do need it to be DESC though :)

Comment (struan) [reply perllib/Open311/GetServiceRequestUpdates.pm]: doh!

**Pull request closed without merging**